src/test/java/org/terracotta/statistics/StatisticMapperTest.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.statistics;
+
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.terracotta.statistics.observer.ChainedOperationObserver;
+
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.EnumSet.of;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.terracotta.statistics.StatisticMapperTest.Source.C;
+import static org.terracotta.statistics.StatisticMapperTest.Source.D;
+import static org.terracotta.statistics.StatisticMapperTest.Source.E;
+import static org.terracotta.statistics.StatisticMapperTest.Target.A;
+import static org.terracotta.statistics.StatisticMapperTest.Target.B;
+
+public class StatisticMapperTest {
+
+  @Test
+  public void testInvalidSourceStatisticSet() {
+    try {
+      new StatisticMapper<Source, Target>(Collections.<Target, Set<Source>>singletonMap(A, of(C, D)), null);
```

Comment:
Ahah! This code has just been moved from ehcache to there. So issues that were there are still there because it has not been touched. See https://goo.gl/NbQCle.

I'll have a look at them ;-) I think there have been some miss in a previous ehcache PR review then!